src/main/java/org/noureddine/joularjx/cpu/PowermetricsMacOS.java

```diff
@@ -61,19 +62,17 @@ public double getCurrentPower(double cpuLoad) {
                 }
 
                 // looking for line fitting the: "<name> Power: xxx mW" pattern and add all of the associated values together
-                final var powerIndicatorIndex = line.indexOf(POWER_INDICATOR);
-
-                // we need an exit condition to avoid looping forever (since there are always new lines, the process being periodical)
-                // if we started processing power lines and we don't find any anymore, we've reached the end of this "page" so exit the loop
-                if(processingPower && powerIndicatorIndex < 0) {
-                    break;
-                }
+                // or, for Intel chips, the: "Intel energy model derived package power (CPUs+GT+SA): xxx W" pattern
+                final var powerIndicatorIndexM = line.indexOf(POWER_INDICATOR_M_CHIP);
+                final var powerIndicatorIndexIntel = line.indexOf(POWER_INDICATOR_INTEL_CHIP);
 
                 // lines with `-` as the second char are disregarded as of the form: "E-Cluster Power: 6 mW" which fits the pattern but shouldn't be considered
                 // also ignore Combined Power if available since it is the sum of the other components
-                if (powerIndicatorIndex >= 0 && '-' != line.charAt(1) && !line.startsWith("Combined")) {
-                    powerInMilliwatts += extractPowerInMilliwatts(line, powerIndicatorIndex);
-                    processingPower = true; // record we're in the power lines section of the powermetrics output
+                if (powerIndicatorIndexM >= 0 && '-' != line.charAt(1) && !line.startsWith("Combined")) {
```

Comment:
Similarly to the previous comment, assuming the CPU type is detected from the headers, this processing should be an either/or situation (though, here, only one of the index should be greater than 0 so the problem of useless processing is minimal).

Reply:
Yes, I've moved intel and m chip parsing to different functions. This allows us to only check for a specific situation and also removed the W / mW conversions.